setuptools_rust/build.py

```diff
@@ -77,9 +79,7 @@ def get_target_info(self):
         # we'll target a 32-bit Rust build.
         # Automatic target detection can be overridden via the CARGO_BUILD_TARGET
         # environment variable or --target command line option
-        if self.target:
-            return _TargetInfo(self.target)
-        elif self.plat_name == "win32":
+        if self.plat_name == "win32":
```

Comment:
Regarding the `FIXME`s: how about separating out the crossenv logic into a `get_crossenv_target_info() -> Option[_TargetInfo]`, and calling that here first:

```suggestion
        crossenv_target_info = self.get_crossenv_target_info()
        
        if crossenv_target_info is not None:
            # TODO: Add some kind of check that self.target matches this target info here,
            # raise `ValueError` if mismatches?
            return crossenv_target_info
        
        if self.target:
            return _TargetInfo(self.target)
        elif self.plat_name == "win32":
```